app/containers/SearchBox.tsx

```diff
@@ -95,6 +106,7 @@ const SearchBox = ({
 				theme={theme!}
 				{...props}
 			/>
+			{hasCancelIcon ? CancelIconButton(onCancelPress!, theme!) : null}
```

Comment:
You need to check both if it is has `hasCancelIcon` prop and if the function `onCancelPress` has been passed
As you haven't declared the default value on the CancelIconButton, using the non-null assertion it can generate bugs as it will try to execute a function that is undefined.

Reply:
Done